### PR TITLE
correctly save form errors

### PR DESCRIPTION
**Pull request by snopoke**
Also validate case field lengths

http://manage.dimagi.com/default.asp?248940#1306368

buddy @mkangia 

best reviewed by commit :fish:

### CODE REVIEW

**Comment by snopoke** [comment]
Still need to do a cleanup of any XForms that were saved but shouldn't have been.

File: corehq/form_processor/tests/test_basic_cases.py [hunk 1]
```diff
@@ -361,7 +364,46 @@ def test_globally_unique_case_id(self):
...
+            owner_id='user1',
+            case_type='demo',
+            case_name='name'
+        )
+
+        try:
+            post_case_blocks([case.as_xml()], domain='domain2')
+            self.fail('expected execption')
+        except DataError:
+            pass
```
Comment (benrudolph) [inline corehq/form_processor/tests/test_basic_cases.py]: nit: assertRaises

**Review by benrudolph — APPROVED** [review]
nice thanks for doing this

File: corehq/form_processor/tests/test_basic_cases.py (hunk 1 above)
Comment (snopoke) [inline corehq/form_processor/tests/test_basic_cases.py]: 7cf0757

**Comment by snopoke** [comment]
In order to handle form edits and other scenarios this is going to be more complex than I thought.

File: corehq/form_processor/backends/sql/update_strategy.py [hunk 2]
```diff
@@ -13,13 +13,30 @@
 from corehq.form_processor.update_strategy_base import UpdateStrategy
 from django.utils.translation import ugettext as _
 
+
+def _validate_length(length):
```
Comment (mkangia) [inline corehq/form_processor/backends/sql/update_strategy.py]: 🆒

**Comment by mkangia** [comment]
@snopoke assuming you would make some more changes to this and tests.

**Pull request closed without merging**